modules/issue_tracker/php/endpoints/attachment.class.inc

```diff
@@ -0,0 +1,245 @@
+<?php declare(strict_types=1);
+
+namespace LORIS\issue_tracker\endpoints;
+
+use \Psr\Http\Message\ServerRequestInterface;
+use \Psr\Http\Message\ResponseInterface;
+use \LORIS\Middleware\ETagCalculator;
+use \LORIS\api\Endpoint;
+
+/**
+ * The Attachment class for requests from issue_trackers.
+ *
+ * @category Loris
+ * @package  IssueTracker
+ * @author   Alizée Wickenheiser <alizee.wickenheiser@mcin.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+class Attachment extends Endpoint implements ETagCalculator
+{
+    public $skipTemplate = true;
+
+    /**
+     * This function will return a json object
+     * for the specific attachment request.
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    public function handle(ServerRequestInterface $request) : ResponseInterface
+    {
+        // Ensure GET, POST, or DELETE request.
+        switch ($request->getMethod()) {
+        case 'GET':
+            return $this->_handleGET($request);
+        case 'POST':
+            return $this->_handlePOST($request);
+        case 'DELETE':
+            return $this->_handleDELETE($request);
+        default:
+            return new \LORIS\Http\Response\JSON\MethodNotAllowed(
+                $this->allowedMethods()
+            );
+        }
+    }
+
+    /**
+     * Handle GET requests for attachment
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request.
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    private function _handleGET(ServerRequestInterface $request) : ResponseInterface
+    {
+        // Parse GET query params.
+        $values = $request->getQueryParams();
+
+        $file_hash = $values['file_hash'];
+        $filename  = $values['filename'];
+        $mime_type = $values['mime_type'];
```

Comment:
@xlecours I'm a little confused. The Content-type header is recommended to be provided when downloading a file of a specific type and for the browser to then handle the type of download. Otherwise I think some browsers behave differently for downloading when it's not given. Can you explain what you mean about the Accept header?

Reply:
resolved offline